SwissTransferFeatures/NewTransferView/VerifyMailView.swift

```diff
@@ -52,12 +72,72 @@ struct VerifyMailView: View {
                 .foregroundStyle(Color.ST.textSecondary)
         }
         .frame(maxHeight: .infinity, alignment: .top)
-        .stNavigationBarNewTransfer()
         .stNavigationBarStyle()
         .padding(value: .medium)
+        .safeAreaButtons {
+            if let error {
+                Text(error.errorDescription)
+                    .font(.ST.caption)
+                    .foregroundStyle(Color.ST.error)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            }
+
+            Button(STResourcesStrings.Localizable.buttonOpenMailApp, action: openMailApp)
+                .buttonStyle(.ikBorderedProminent)
+                .disabled(isVerifyingCode)
+
+            ResendCodeButton(emailToVerify: newUploadSession.authorEmail, resendTimeDelaySeconds: 30, error: $error)
+                .disabled(isVerifyingCode)
+        }
+    }
+
+    func openMailApp() {
+        guard let openMailURL = URL(string: "message:") else { return }
+        openURL(openMailURL)
+    }
+
+    func verifyCode(_ code: String) {
+        guard !isVerifyingCode else { return }
+        isVerifyingCode = true
+
+        Task {
+            do {
+                let verifyEmailCodeBody = STNVerifyEmailCodeBody(code: code, email: newUploadSession.authorEmail)
+                let token = try await STNUploadRepository().verifyEmailCode(verifyEmailCodeBody: verifyEmailCodeBody).token
```

Comment:
Should `STNUploadRepository` be injected with SwissTransferInjection or we don't care?

Reply:
I think it wasn't available in the beginning. Now I can use it 👌